src/main/java/com/apuope/apuope_re/services/ChatbotService.java

```diff
@@ -0,0 +1,118 @@
+import java.util.Optional;
+
+@Service
+public class ChatbotService {
+    private final DSLContext dslContext;
+    private final JWTService jwtService;
+    private final ConversationRepository conversationRepository;
+    private final UserRepository userRepository;
+    private final RestTemplate restTemplate = new RestTemplate();
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Value("${llm.api.url}")
+    private String apiUrl;
+
+    @Value("${llm.api.key}")
+    private String apiKey;
+
+    public ChatbotService(DSLContext dslContext, JWTService jwtService, ConversationRepository conversationRepository,
+            UserRepository userRepository) {
+        this.dslContext = dslContext;
+        this.jwtService = jwtService;
+        this.conversationRepository = conversationRepository;
+        this.userRepository = userRepository;
+    }
+
+    public List<ConversationData> fetchAllConversations(String token, HttpServletRequest request) {
+        String userEmail = jwtService.extractEmail(token);
+        Optional<UsersRecord> userOpt = userRepository.findVerifiedUserByEmail(userEmail, dslContext);
+        if (userOpt.isPresent()) {
+            return conversationRepository.fetchConversationByAccountId(userOpt.get().getId(), dslContext);
+        }
+        return Collections.emptyList();
+    }
+
+    public List<MessageData> fetchConversation(Integer conversationId) {
+        return conversationRepository.fetchConversationById(conversationId, dslContext);
+    }
+
+    public ConversationRecord startConversation(Integer userId, ChatRequestData request) {
+        return conversationRepository.createConversation(userId, request.getChapterId(), "", dslContext);
+    }
+
+    public ResponseData<MessageData> sendRequest(
+            String token, HttpServletRequest request,
+            ChatRequestData chatRequest
+    ) throws JsonProcessingException {
+        Integer conversationId;
+
+        String userEmail = jwtService.extractEmail(token);
+        Optional<UsersRecord> userOpt = userRepository.findVerifiedUserByEmail(userEmail, dslContext);
+
+        if (userOpt.isPresent() && (chatRequest.getConversationId() == 0 || chatRequest.getConversationId() == null)) {
+            ConversationRecord conversation = startConversation(userOpt.get().getId(), chatRequest);
+            conversationId = conversation.getId();
+
+            conversationRepository.createMessage(conversationId, chatRequest.getData(), MessageSource.USER.getValue(),
+                    dslContext);
+        } else {
+            conversationId = chatRequest.getConversationId();
+            conversationRepository.createMessage(chatRequest.getConversationId(), chatRequest.getData(), MessageSource.USER.getValue(),
+                    dslContext);
+        }
+
+        // Comment out lines 93-107 and uncomment line 108 if you can't connect to LLM.
+        // Line 108 will sent mocked answer every time message is sent to LLM.
```

Comment:
Do we need this mock thing not when vpn works. We don't want this to production anyway?

Reply:
This PR was implemented before VPN worked, so no, we don't need this anymore :)